File: reverse_integer.rb
# @param {Integer} x
# @return {Integer}
def reverse(x)
  if x == 0
    0
  elsif x > 0
    y = x.to_s.reverse.to_i
    y > 2**31 - 1 ? 0 : y
  else
    y = -(x.abs.to_s.reverse.to_i)
    y < -2**31 + 1 ? 0 : y
  end
end

File: reverse_integer.py
class Solution:
  def reverse(self, x):
    arr = list(str(abs(x)))
    arr.reverse()
    y = int(''.join(arr))
    if x < 0:
      y = -y
      return 0 if y < -2**31 + 1 else y
    else:
      return 0 if y > 2**31 - 1 else y